cluster/local/integration_tests_tls.sh

```diff
@@ -0,0 +1,354 @@
+openssl req -new -key client-key.pem -out client-req.pem -subj "/CN=client"
+openssl x509 -req -in client-req.pem -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out client-cert.pem -days 365
+
+# Create a secret for the server TLS certificates and keys
+#"${KUBECTL}" delete secret mariadb-server-tls || true
+"${KUBECTL}" create secret generic mariadb-server-tls \
+    --from-file=ca-cert.pem \
+    --from-file=server-cert.pem \
+    --from-file=server-key.pem
+
+# Create a secret for the client TLS certificates and keys
+#"${KUBECTL}" delete secret mariadb-client-tls || true
+"${KUBECTL}" create secret generic mariadb-client-tls \
+    --from-file=ca-cert.pem \
+    --from-file=client-cert.pem \
+    --from-file=client-key.pem
+
+# install MariaDB chart with TLS enabled
+echo_step "installing MariaDB Helm chart with TLS into default namespace"
+mariadb_root_pw=$(openssl rand -base64 32)
+mariadb_test_pw=$(openssl rand -base64 32)
+
+"${HELM3}" repo add bitnami https://charts.bitnami.com/bitnami
+
+MARIADB_VALUES=$(cat <<EOF
+auth:
+  rootPassword: ${mariadb_root_pw}
+primary:
+  extraFlags: "--ssl --require-secure-transport=ON --ssl-ca=/opt/bitnami/mariadb/certs/ca-cert.pem --ssl-cert=/opt/bitnami/mariadb/certs/server-cert.pem --ssl-key=/opt/bitnami/mariadb/certs/server-key.pem"
+  configurationSecret: mariadb-server-tls
+  extraVolumes:
+    - name: tls-certificates
+      secret:
+        secretName: mariadb-server-tls
+  extraVolumeMounts:
+    - name: tls-certificates
+      mountPath: /opt/bitnami/mariadb/certs
+      readOnly: true
+initdbScripts:
+  init.sql: |
+    CREATE USER 'test'@'%' IDENTIFIED BY '${mariadb_test_pw}' REQUIRE X509;
+    GRANT ALL PRIVILEGES ON *.* TO 'test'@'%' WITH GRANT OPTION;
+    FLUSH PRIVILEGES;
+EOF
+)
+
+# install MariaDB chart
+"${HELM3}" install mariadb bitnami/mariadb \
+    --version 11.3.0 \
+    --values <(echo "$MARIADB_VALUES") \
+    --wait
+
+# create ProviderConfig with TLS
+"${KUBECTL}" create secret generic mariadb-creds \
+    --from-literal username="test" \
+    --from-literal password="${mariadb_test_pw}" \
+    --from-literal endpoint="mariadb.default.svc.cluster.local" \
+    --from-literal port="3306" \
+    --from-file=ca-cert.pem \
+    --from-file=client-cert.pem \
+    --from-file=client-key.pem
+
+PROVIDER_CONFIG_YAML="$( cat <<EOF
+apiVersion: mysql.sql.crossplane.io/v1alpha1
+kind: ProviderConfig
+metadata:
+  name: default
+spec:
+  credentials:
+    source: MySQLConnectionSecret
+    connectionSecretRef:
+      namespace: default
+      name: mariadb-creds
+  tls: custom
+  tlsConfig:
+    caCert:
+      secretRef:
+        namespace: default
+        name: mariadb-creds
+        key: ca-cert.pem
+    clientCert:
+      secretRef:
+        namespace: default
+        name: mariadb-creds
+        key: client-cert.pem
+    clientKey:
+      secretRef:
+        namespace: default
+        name: mariadb-creds
+        key: client-key.pem
+    insecureSkipVerify: true
```

Comment:
`insecureSkipVerify` is to be used only here in e2e tests, because certs are self-signed, otherwise server would reject its own cert.